apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/R4ExplanationOfBenefitResourceProviderIT.java

```diff
@@ -0,0 +1,1769 @@
+    return fhirClient.search().byUrl(url).returnBundle(Bundle.class).execute();
+  }
+
+  /*
+   * Verify that each of the expected claims (one for every claim type) is present
+   * and looks correct.
+   */
+  private static void assertEachEob(Bundle searchResults, List<Object> loadedRecords) {
+    compareEob(ClaimTypeV2.CARRIER, searchResults, loadedRecords);
+    compareEob(ClaimTypeV2.DME, searchResults, loadedRecords);
+    compareEob(ClaimTypeV2.HHA, searchResults, loadedRecords);
+    compareEob(ClaimTypeV2.HOSPICE, searchResults, loadedRecords);
+    compareEob(ClaimTypeV2.INPATIENT, searchResults, loadedRecords);
+    compareEob(ClaimTypeV2.OUTPATIENT, searchResults, loadedRecords);
+    compareEob(ClaimTypeV2.PDE, searchResults, loadedRecords);
+    compareEob(ClaimTypeV2.SNF, searchResults, loadedRecords);
+  }
+
+  /**
+   * Asserts that two Money values ignoring differences like "0" vs "0.0"
+   *
+   * @param expected
+   * @param actual
+   */
+  private static void assertEquals(Money expected, Money actual) {
+    Assert.assertEquals(expected.getCurrency(), actual.getCurrency());
+
+    // Money will return null instead of auto creating value
+    if (expected.hasValue()) {
+      Assert.assertTrue(actual.hasValue());
+      Assert.assertEquals(expected.getValue().floatValue(), actual.getValue().floatValue(), 0.0);
+    } else {
+      // If expected has no value, then actual can't either
+      Assert.assertFalse(actual.hasValue());
+    }
+  }
+
+  /**
+   * Compares two ExplanationOfBenefit objects in detail while working around serialization issues
+   * like comparing "0" and "0.0" or creation differences like using "Quantity" vs "SimpleQuantity"
+   *
+   * @param expected
+   * @param actual
+   */
+  private static void assertEquals(ExplanationOfBenefit expected, ExplanationOfBenefit actual) {
+    // ID in the bundle will have `ExplanationOfBenefit/` in front, so make sure the last bit
+    // matches up
+    Assert.assertTrue(actual.getId().endsWith(expected.getId()));
+
+    // Clean them out so we can do a deep compare later
+    actual.setId("");
+    expected.setId("");
+
+    // If there are any contained resources, they might have lastupdate times in them too
+    Assert.assertEquals(expected.getContained().size(), actual.getContained().size());
+    for (int i = 0; i < expected.getContained().size(); i++) {
+      Resource expectedContained = expected.getContained().get(i);
+      Resource actualContained = actual.getContained().get(i);
+
+      expectedContained.getMeta().setLastUpdated(null);
+      actualContained.getMeta().setLastUpdated(null);
+
+      // TODO: HAPI seems to be inserting the `#` in the ID of the contained element.
+      // This is incorrect according to the FHIR spec:
+      // https://build.fhir.org/references.html#contained
+      // This works around that problem
+      Assert.assertEquals("#" + expectedContained.getId(), actualContained.getId());
```

Comment:
The FHIR spec suggests that the actual contained resource should *not* have an ID that starts with `#`.  Our transformers do not add this.  It appears that HAPI is adding it when you go through the actual read endpoint.  To get around this in testing, we prefix the expected ID with `#` as well.  This is not correct as far as I can tell and may be a bug with HAPI.

Reference: https://build.fhir.org/references.html#contained